### DISCUSSION TITLE
How can I forward span to service using opentracing

### DISCUSSION THREAD

**Discussion by zhsj** [open-telemetry/opentelemetry-go#2475] [Q&A] [answered]
I have a service A using otel, and a service B(which I can't modify its source) using opentracing.
In service A, it sends gRPC request to service B.

In service A, I already has

```go
        opts := []grpc.DialOption{
                grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
                grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor()),
        }
        conn, err := client.Dial(addr, opts)
```

But I don't know how to propagate the tracing info from A to B.

**Comment by MrAlias** [comment, marked as answer]
You need to propagate context using a propagator. First you'll need to determine what propagation format (W3C tracecontext, B3, ot ...) the receiving service is using and then use an appropriate propagator from [the main repo](https://pkg.go.dev/go.opentelemetry.io/otel@v1.3.0/propagation) or [contrib](https://pkg.go.dev/go.opentelemetry.io/contrib/propagators).

**Reply by zhsj** [reply]
Thanks. `go.opentelemetry.io/contrib/propagators` is definitely what I'm missing.

**Comment by rogierlommers** [comment]
Hey, I'm having a similar issue and I cannot figure out how to fix this. I have set-up a repo to illustrate my issue: https://github.com/rogierlommers/otel-b3-header-propagation

Can someone point me to some advice how to fix the B3 header propagation?
